test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatLocalsTests.cs

```diff
@@ -26,19 +29,50 @@ public class XPlatLocalsTests
         [InlineData("locals global-packages -l")]
         [InlineData("locals --list global-packages")]
         [InlineData("locals -l global-packages")]
-        // [InlineData("locals --clear all")]
-        // [InlineData("locals -c all")]
-        public static void Locals_Succeeds(String args)
+        [InlineData("locals --clear all")]
+        [InlineData("locals -c all")]
+        [InlineData("locals http-cache --clear")]
+        [InlineData("locals http-cache -c")]
+        [InlineData("locals --clear http-cache")]
+        [InlineData("locals -c http-cache")]
+        [InlineData("locals temp --clear")]
+        [InlineData("locals temp -c")]
+        [InlineData("locals --clear temp")]
+        [InlineData("locals -c temp")]
+        [InlineData("locals global-packages --clear")]
+        [InlineData("locals global-packages -c")]
+        [InlineData("locals --clear global-packages")]
+        [InlineData("locals -c global-packages")]
+        public static void Locals_Succeeds(string args)
         {
-            // Arrange
-            var log = new TestCommandOutputLogger();
+            using (var mockBaseDirectory = TestFileSystemUtility.CreateRandomTestFolder())
+            {
+                // Arrange
+                var mockGlobalPackagesDirectory = Directory.CreateDirectory(Path.Combine(mockBaseDirectory.Path, @"global-packages"));
+                var mockHttpCacheDirectory = Directory.CreateDirectory(Path.Combine(mockBaseDirectory.Path, @"http-cache"));
+                var mockTmpCacheDirectory = Directory.CreateDirectory(Path.Combine(mockBaseDirectory.Path, @"temp"));
 
-            // Act
-            var exitCode = Program.MainInternal(args.Split(null), log);
+                DotnetCliUtil.CreateTestFiles(mockGlobalPackagesDirectory.FullName);
+                DotnetCliUtil.CreateTestFiles(mockHttpCacheDirectory.FullName);
+                DotnetCliUtil.CreateTestFiles(mockTmpCacheDirectory.FullName);
 
-            // Assert
-            Assert.Equal(string.Empty, log.ShowErrors());
-            Assert.Equal(0, exitCode);
+                // Act
+                var result = CommandRunner.Run(
+                      DotnetCliUtil.GetDotnetCli(),
+                      Directory.GetCurrentDirectory(),
+                      DotnetCliUtil.GetXplatDll() + " " + args,
+                      waitForExit: true,
+                    environmentVariables: new Dictionary<string, string>
+                    {
+                        { "NUGET_PACKAGES", mockGlobalPackagesDirectory.FullName },
+                        { "NUGET_HTTP_CACHE_PATH", mockHttpCacheDirectory.FullName },
+                        { RuntimeEnvironmentHelper.IsWindows?"TMP":"TMPDIR", mockTmpCacheDirectory.FullName }
+                    });
+                // Unix uses TMPDIR as environment variable as opposed to TMP on windows
+
+                // Assert
+                DotnetCliUtil.VerifyResultSuccess(result, string.Empty);
```

Comment:
`!Directory.Exists` and `Directory.EnumerateFileSystemEntries` recursive count should be 0.